src/vocabulary_builder.py

```diff
@@ -45,35 +49,43 @@ def _(language: str):
 def fetch_random_word_data(db: Session):
     random_row = get_random_word(db)
     data = {
-        'word': random_row.word,
-        'translated_word': random_row.translated_word,
-        'context': random_row.context,
-        'translated_context': random_row.translated_context,
+        "word": random_row.word,
+        "translated_word": random_row.translated_word,
+        "context": random_row.context,
+        "translated_context": random_row.translated_context,
     }
     return data
 
 
-@app.get('/', response_class=HTMLResponse)
-def get_main_page(request: Request, db: Session = Depends(get_db)):
-   return handle_main_page(request, db)
+@app.get("/", response_class=HTMLResponse)
+def get_main_page(request: Request, db: Optional[Session] = None):
+    if db is None:
+        db = next(get_db())
+    return handle_main_page(request, db)
 
 
-@app.get('/{language}', response_class=HTMLResponse)
-def get_main_page_in_language(request: Request, language: str, db: Session = Depends(get_db)):
+@app.get("/{language}", response_class=HTMLResponse)
+def get_main_page_in_language(
+    request: Request, language: str, db: Optional[Session] = None
+):
+    if db is None:
+        db = next(get_db())
     return handle_main_page(request, db, language)
 
 
-def handle_main_page(request: Request, db: Session, language: str = 'ru'):
+def handle_main_page(request: Request, db: Session, language: str = "ru"):
     context = fetch_random_word_data(db)
-    context.update({'_': _(language)})
+    context.update({"_": _(language)})
     return templates.TemplateResponse(
         request=request,
-        name='index.html',
+        name="index.html",
         context=context,
     )
 
 
-@app.get('/new_word', response_class=JSONResponse)
-def get_new_word(db: Session = Depends(get_db)):
+@app.get("/new_word", response_class=JSONResponse)
+def get_new_word(db: Optional[Session] = None):

```

Comment:
It's better to ignore flake here and use `Depends`

Reply:
Done